File: Golang/fizzbuzz.go
package main

import "fmt"

func main() {
	for i := 1; i <= 100; i++ {
		result := ""
		if i%3 == 0 {
			result += "Fizz"
		}
		if i%5 == 0 {
			result += "Buzz"
		}
		if result != "" {
			fmt.Println(result)
			continue
		}
		fmt.Println(i)
	}
}

File: Rust/fizzbuzz.rs
// My take on fizzbuzz.
// Author: Ethan Brierley

fn main() {
    let fizzbuzz = (1..).map(|val| {
        if val % 3 == 0 {
            if val % 5 == 0 {
                "fizzbuzz".to_string()
            } else {
                "fizz".to_string()
            }
        } else if val % 5 == 0 {
            "buzz".to_string()
        } else {
            val.to_string()
        }
    });
    for x in fizzbuzz.take(25) {
        println!("{}", x)
    }
}
